src/hsm.rs

```diff
@@ -199,36 +212,49 @@ pub fn reset(client: &Client) -> Result<()> {
 /// uses these shares to reconstitute the wrap key. This wrap key can then
 /// be used to restore previously backed up / export wrapped keys.
 pub fn restore_wrap(client: &Client) -> Result<()> {
-    let mut shares: Vec<String> = Vec::new();
+    let mut shares: Vec<[u8; KEY_LEN + 1]> = Vec::new();
 
     for i in 1..=THRESHOLD {
         print!("Enter share[{}]: ", i);
         io::stdout().flush()?;
-        shares.push(io::stdin().lines().next().unwrap().unwrap());
+        shares.push(
+            hex::decode(io::stdin().lines().next().unwrap().unwrap())?
+                .try_into()
+                .map_err(|_: _| anyhow::anyhow!("malformed keyshare"))?,
```

Comment:
.with_context() ?

Reply:
it was getting late last night :smiling_face_with_tear:

Reply:
oh I spent a bit wrestling with this and punted with the anyhow approach after a bit. This boiled down to the `try_into()` returning `Result<[u8; 33], Vec<u8>>` in this case. This breaks `with_context`. The "right thing to do" seems to be creating a new variant in the `HsmError` enum.